frontend/src/components/tabs/MetricsTab.test.tsx

```diff
@@ -0,0 +1,208 @@
+              recall: 0,
+            },
+            {
+              confidenceThreshold: 1,
+              falsePositiveRate: 0,
+              recall: 0.33962264150943394,
+            },
+          ],
+        }),
+      ),
+    );
+
+    jest.spyOn(mlmdUtils, 'getOutputArtifactsInExecution').mockResolvedValue([artifact]);
+    jest.spyOn(mlmdUtils, 'getArtifactTypes').mockResolvedValue([artifactType]);
+
+    console.log('shows error banner when confidenceMetric type is wrong');
+    const { getByText } = render(
+      <CommonTestWrapper>
+        <MetricsTab execution={execution}></MetricsTab>
+      </CommonTestWrapper>,
+    );
+
+    await waitFor(() => getByText("Error in metrics artifact's confidenceMetrics data format."));
+  });
+
+  it('shows error banner when confidenceMetric is not array', async () => {
+    const execution = buildBasicExecution().setLastKnownState(Execution.State.COMPLETE);
+    const artifactType = buildBasicArtifactType();
+    const artifact = buildBasicArtifact();
+    artifact.getCustomPropertiesMap().set('name', new Value().setStringValue('metrics'));
+    artifact.getCustomPropertiesMap().set(
+      'confidenceMetrics',
+      new Value().setStructValue(
+        Struct.fromJavaScript({
+          variable: 'i am not a list',
+        }),
+      ),
+    );
+
+    jest.spyOn(mlmdUtils, 'getOutputArtifactsInExecution').mockResolvedValue([artifact]);
+    jest.spyOn(mlmdUtils, 'getArtifactTypes').mockResolvedValue([artifactType]);
+
+    console.log('shows error banner when confidenceMetric type is wrong');
+    const { getByText } = render(
+      <CommonTestWrapper>
+        <MetricsTab execution={execution}></MetricsTab>
+      </CommonTestWrapper>,
+    );
+
+    await waitFor(() => getByText("Error in metrics artifact's confidenceMetrics data format."));
+  });
+
+  it('has no metrics', async () => {
+    jest.spyOn(mlmdUtils, 'getOutputArtifactsInExecution').mockReturnValue(Promise.resolve([]));
+    jest.spyOn(mlmdUtils, 'getArtifactTypes').mockReturnValue(Promise.resolve([]));
+    const execution = buildBasicExecution().setLastKnownState(Execution.State.COMPLETE);
+
+    const { getByText } = render(
+      <CommonTestWrapper>
+        <MetricsTab execution={execution}></MetricsTab>
+      </CommonTestWrapper>,
+    );
+
+    await waitFor(() => getByText('There is no metrics artifact available in this step.'));
+  });
+
+  it('shows info banner when execution is in UNKNOWN', async () => {
+    jest.spyOn(mlmdUtils, 'getOutputArtifactsInExecution').mockReturnValue(Promise.resolve([]));
+    jest.spyOn(mlmdUtils, 'getArtifactTypes').mockReturnValue(Promise.resolve([]));
+
+    const execution = buildBasicExecution().setLastKnownState(Execution.State.UNKNOWN);
+    const { getByText } = render(
+      <CommonTestWrapper>
+        <MetricsTab execution={execution}></MetricsTab>
+      </CommonTestWrapper>,
+    );
+    await waitFor(() => getByText('Task has not completed.'));
```

Comment:
NEW and RUNNING are expected, this UNKNOWN phase should ideally have a different text, because it's unexpected: 'Task is in unknown state.'. WDUT?

Reply:
That makes sense, I have added a condition to check unknown state for this banner message.